src/garage/tf/algos/rl2.py

```diff
@@ -185,9 +332,18 @@ def _process_samples(self, itr, paths):
             np_tensor_utils.stack_and_pad_tensor_dict_list(
                 concatenated_paths, self._inner_algo.max_path_length))
 
-        undiscounted_returns = log_performance(
-            itr, TrajectoryBatch.from_trajectory_list(self._env_spec, paths),
-            self._inner_algo.discount)
+        name_map = None
+        if hasattr(self._task_sampler._envs[0].env, 'all_task_names'):
```

Comment:
@ryanjulian @krzentner Does this part look ok to you?

Reply:
hmm would it be simpler to just check if it's from metaworld? Do we expect anyone else to be implementing that interface?

Reply:
I'm not terribly offended by this, but I think we should file an issue about cleaning it up. It seems there's a missing interface here for multitask, e.g. something like this could go into a multitask version of GarageEnv.

Reply:
I like the idea of a multitask version of GarageEnv. So far in docstring, I write `GarageEnv` but what I truly want is a `GarageMultiTaskEnv`.